controllers/daemon/staticgatewayconfiguration_controller.go

```diff
@@ -315,64 +311,25 @@ func (r *StaticGatewayConfigurationReconciler) getVMIP(
 	gwConfig *egressgatewayv1alpha1.StaticGatewayConfiguration,
 ) (string, string, error) {
 	log := log.FromContext(ctx)
-	matches := vmssInstanceRE.FindStringSubmatch(nodeMeta.Compute.ResourceID)
-	if len(matches) != 5 {
-		return "", "", fmt.Errorf("failed to parse vmss instance resource ID: %s", nodeMeta.Compute.ResourceID)
-	}
-	subscriptionID, resourceGroupName, vmssName, instanceID := matches[1], matches[2], matches[3], matches[4]
-	if subscriptionID != r.SubscriptionID() {
-		return "", "", fmt.Errorf("node subscription(%s) is different from configured subscription(%s)", subscriptionID, r.SubscriptionID())
-	}
-	vm, err := r.GetVMSSInstance(ctx, resourceGroupName, vmssName, instanceID)
-	if err != nil {
-		return "", "", fmt.Errorf("failed to get vmss instance: %w", err)
-	}
-	if vm.Properties == nil || vm.Properties.NetworkProfileConfiguration == nil {
-		return "", "", fmt.Errorf("vm has empty network properties")
-	}
 
-	ipConfigName := gwConfig.Namespace + "_" + gwConfig.Name
-	interfaces := vm.Properties.NetworkProfileConfiguration.NetworkInterfaceConfigurations
-	nicName := ""
-	for _, nic := range interfaces {
-		if nic.Properties != nil && to.Val(nic.Properties.Primary) {
-			nicName = to.Val(nic.Name)
+	nodeName := nodeMeta.Compute.OSProfile.ComputerName
+	var primaryIP, secondaryIP string
+
+	for _, vmProfile := range gwConfig.Status.GatewayVMProfiles {
```

Comment:
What if daemon directly reads the gatewayVMConfiguration object? gatewayVMConfiguration has the same ns/name as the staticGatewayConfiguration obj. In this case, no need to pass this information all the way back to gatewayLBConfiguration and staticGatewayConfiguration?